### PR TITLE
Adding `prettier` tool

### PR DESCRIPTION
**Pull request by jamesbraza**
I noticed the docs had extra whitespace. This PR add `prettier` and `end-of-file-fixer` to `pre-commit` for standardized spacing

### CODE REVIEW

**Review by RonnyPfannschmidt — CHANGES_REQUESTED** [review]
im happy to add some fixes, but i am well aware why prettier is no longer under pre-commit and i will not add that tool unless the pre-commit maintainers are convinced it wont mess up everything again

i was at the end of said breakages multiple times myself and consider it scorched earth for now

**Comment by jamesbraza** [comment]
Fwiw this PR uses https://github.com/rbubley/mirrors-prettier (as opposed to the archived https://github.com/pre-commit/mirrors-prettier). Regardless, I reverted the addition of `prettier` to `.pre-commit-config.yaml`.

Are you okay with the changes `prettier` makes? Also, feel free to just close this PR if you're not interested at all, I am not attached to it.

**Comment by jamesbraza** [comment]
Alright I just rebased this atop `main` 🤓 . Note there is no `mirrors-prettier` here in this PR, just the changes made by the tool, as if it was a ghost 👻

File: docs/extending.md [hunk 1]
```diff
@@ -9,63 +9,59 @@
...
 
+Also uses then name of your SCM control directory as name of the entrypoint.
 
 ### api reference for scm version objects
 
 ::: setuptools_scm.version.ScmVersion
-    options:
-      show_root_heading: yes
-      heading_level: 4
+options:
```
Comment (RonnyPfannschmidt) [inline docs/extending.md:30]: its broken

Comment (jamesbraza) [reply docs/extending.md:30]: Oh dang, too bad. I made https://github.com/prettier/prettier/issues/16713

I will close this PR as this seems to be a critical bug

**Pull request closed without merging**